src/app/project-report/project-report.page.ts

```diff
@@ -0,0 +1,297 @@
+import { Component, OnInit, AfterViewInit, ViewChild, inject } from '@angular/core';
+import { NavController } from '@ionic/angular';
+import { LoaderService } from '../services/loader/loader.service';
+import { ApiBaseService } from '../services/base-api/api-base.service';
+import urlConfig from 'src/app/config/url.config.json';
+import { UrlConfig } from '../interfaces/main.interface';
+import { finalize } from 'rxjs';
+import { ToastService } from '../services/toast/toast.service';
+import { Chart, registerables } from 'chart.js';
+import ChartDataLabels from 'chartjs-plugin-datalabels';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-project-report',
+  templateUrl: './project-report.page.html',
+  styleUrls: ['./project-report.page.scss'],
+})
+export class ProjectReportPage implements OnInit {
+  isFilter: boolean = false;
+  loader: LoaderService;
+  baseApiService: any;
+  listType!: keyof UrlConfig;
+  toastService: ToastService;
+  reportData: any;
+  reportType: string = "1";
+  programId: string = "";
+  isModalOpen = false;
+  selectedProgram: string = "";
+  programList: any;
+  projectsArr:any;
+
+
+  constructor(
+    private navCtrl: NavController,
+    private router: ActivatedRoute
+  ) {
+    this.loader = inject(LoaderService);
+    this.baseApiService = inject(ApiBaseService);
+    this.toastService = inject(ToastService);
+    Chart.register(...registerables);
+    Chart.register(ChartDataLabels);
+  }
+
+  ngOnInit() {
+    this.listType = 'report';
+    this.getReportData(this.reportType);
+     this.projectsArr = [
+      {
+        name: 'Total Projects',
+        img: '/assets/images/report-imgs/Note 1.svg',
+        key: 'total',
+      },
+      {
+        name: 'Projects submitted',
+        img: '/assets/images/report-imgs/note.svg',
+        key: 'submitted',
+      },
+      {
+        name: 'Projects inProgress',
+        img: '/assets/images/report-imgs/Note 4.svg',
+        key: 'inProgress',
+      },
+      {
+        name: 'Projects started',
+        img: '/assets/images/report-imgs/Note 3.svg',
+        key: 'started',
+      },
+    ];
+    this.getPrograms();
+    setTimeout(() => {
+      this.renderChart(this.reportData?.tasks, this.reportData?.categories);
+    });
+  }
+
+  goBack() {
+    this.navCtrl.back();
+  }
+
+  getReportType(e: any) {
+    this.reportType = e;
+    this.getReportData(this.reportType);
+  }
+
+  setOpen(isOpen: boolean) {
+    this.isModalOpen = isOpen;
+  }
+
+  async getReportData(reportType: string) {
+    await this.loader.showLoading('Please wait while loading...');
+    if (!this.programId) {
```

Comment:
Why are we using this condition and repeating the same code in both the conditions?

Reply:
Based on the programId only the url should be modified right, the rest all flow remains the same right

Reply:
there are different api's for calling data. if there is a api when we have a programID and other if there is no programID.

Reply:
for now i have commented out the code later when we have the api will be fixing that